admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/rest/ApplicationUploadEndpointTest.java

```
@@ -0,0 +1,114 @@
+
+import org.apache.cxf.jaxrs.ext.multipart.Attachment;
+import org.apache.cxf.jaxrs.ext.multipart.ContentDisposition;
+import org.apache.cxf.jaxrs.ext.multipart.MultipartBody;
+import org.codice.ddf.admin.application.service.ApplicationService;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ApplicationUploadEndpointTest {
+    private Logger logger = LoggerFactory.getLogger(ApplicationUploadEndpoint.class);
+
+    /**
+     * Tests the {@link ApplicationUploadEndpoint#update(MultipartBody, UriInfo)} method
+     */
+    @Test
+    public void testApplicationUploadEndpointUpdate() {
+        ApplicationService testAppService = mock(ApplicationService.class);
+        MultipartBody testMultipartBody = mock(MultipartBody.class);
+        UriInfo testUriInfo = mock(UriInfo.class);
+        List<Attachment> attachmentList = new ArrayList<>();
+        Attachment testAttach1 = mock(Attachment.class);
+        attachmentList.add(testAttach1);
+        ContentDisposition testDisp = mock(ContentDisposition.class);
+        DataHandler testDataHandler = mock(DataHandler.class);
+
+        when(testAttach1.getDataHandler()).thenReturn(testDataHandler);
+        when(testAttach1.getContentDisposition()).thenReturn(testDisp);
+        when(testMultipartBody.getAllAttachments()).thenReturn(attachmentList);
+
+        try {
+            File testFile = new File(File.class.getResource("/test-kar.zip").getPath());
+            InputStream testIS = new FileInputStream(testFile);
+            when(testDataHandler.getInputStream()).thenReturn(testIS);
+
+            ApplicationUploadEndpoint applicationUploadEndpoint = new ApplicationUploadEndpoint(
+                    testAppService);
+
+            applicationUploadEndpoint
+                    .setDefaultFileLocation("target/ApplicationUploadEndpointTest");
+            assertNotNull(applicationUploadEndpoint.update(testMultipartBody, testUriInfo));
```

What type of response are you expecting here? Could assert that it's a success/fail/whatever instead of just not null.

For future reference we tend to like assertThat() better as well, since it produces much more readable test output.